src/atlas_protocol_scripts/pp/_check_genes.py

```diff
@@ -0,0 +1,80 @@
+import re
+from typing import List
+
+import anndata
+import numpy as np
+import pandas as pd
+
+
+def remove_gene_version(string: str) -> str:
+    """Remove gene version from ensembl gene id that start with "ENSG"."""
+    string = str(string)
+    if string.startswith("ENSG"):
+        return re.sub(r"\..*", "", string)
+    else:
+        return string
+
+
+def append_duplicate_suffix(df: pd.DataFrame, column: str, sep: str) -> pd.DataFrame:
+    """Appends a numeric suffix to each duplicated value in the specified column based on index position."""
+    suffix_dict = {}
+    df_sorted = df.sort_values(by=column)
+    df_sorted = df_sorted.reset_index().sort_values(by="index").set_index("index")
+    for i, val in enumerate(df_sorted[column]):
+        if val in suffix_dict:
+            suffix_dict[val] += 1
+            df_sorted.at[df_sorted.index[i], column] = f"{val}{sep}{suffix_dict[val]}"
+        else:
+            suffix_dict[val] = 0
+    return df_sorted.sort_index()
+
+
+def find_unmapped_genes(adata: anndata.AnnData) -> List[str]:
+    """Finds genes in the specified AnnData object that are not mapped to any ensembl id."""
+    unmapped = []
+    for column in ["Gene", "symbol"]:
```

Comment:
not sure it makes sense to hardcode the column names here? what if a dataset has other columns that contain gene names, e.g. `SYMBOL`, `GeneSymbol`, ...

possible solution is to specify columns as parameters. "Explicit is better than implicit"